SPARCore/GADArticle.m

```diff
@@ -1,115 +1,98 @@
+static NSString *const API_KEY_ARTICLE_ID = @"id";
+static NSString *const API_KEY_PUBLICATION_ID = @"publication";
+static NSString *const API_KEY_TITLE = @"title";
 
-- (void) fetchFulltextWithCompletion: (void(^_Nonnull)(GADRemoteModel * *_Nullable model,
-                                                NSError *_Nullable error))completion {
-    NSURL *queryURL = [GADArticle createURLWithArticle:self.articleId publication:self.publicationId];
-    
-    [GADRemoteModel fetchModelsWithParams:queryURL
-                          queryParameters:@{}
-                         modelTransformer:^(NSData *jsonData){
-                             return [GADArticle articlesFromJSON:jsonData];
-                         }
-                        completionHandler:^(NSArray <GADArticle *> *_Nullable model, NSError *_Nullable error){
-                            GADArticle *fullArticle = model[0];
-                            self.content = fullArticle.content;
-                            self.authors = fullArticle.authors;
-                            //completion(); --How should we call the completion handler here?
-                        }];
-    
-}
+static NSString *const API_QUERY_PAGE_TOKEN = @"pageToken";
 
-+ (NSArray <GADArticle *> *) articlesFromJSON: (NSData *)json {
++ (NSArray <GADArticle *> *) articlesFromArray: (NSArray *)jsonArray {
     
-    NSMutableArray <GADArticle *> *articles = [[NSMutableArray alloc] init];
-    NSError *error = nil;
-    NSArray *jsonArray = [NSJSONSerialization JSONObjectWithData:json
-                                                         options:kNilOptions error:&error];
-    
-    if (error != nil) {
-        NSLog(@"Error parsing JSON.");
-    }
+    NSMutableArray <GADArticle *> *articles = [NSMutableArray new];
     
     for (NSDictionary *element in jsonArray) {
-        //Map fields of element to fields of article
-        GADArticle *article = [[GADArticle alloc] init];
-        //datePublished field is a UNIX Timestamp number - converting to NSDate here
-        int timeStamp = (int)element[API_DATE_PUBLISHED];
-        article.datePublished = [NSDate dateWithTimeIntervalSince1970: timeStamp];
-        article.brief = element[API_BRIEF];
-        article.headerImage = element[API_HEADER_IMAGE];
-        article.publicationId = element[API_PUBLICATION_ID];
-        article.articleId = element[API_ARTICLE_ID];
-        article.title = element[API_TITLE];
-        if (element[API_CONTENT]) {
-            article.content = element[API_CONTENT];
-            article.authors = element[API_AUTHORS];
-        }
+        GADArticle *article = [self articleFromDictionary:element];
         [articles addObject:article];
     }
     return articles;
 }
 
-+ (NSURL *) baseURL {
-    NSURL *queryURL = [NSURL URLWithString:API_HOSTNAME];
-    queryURL = [NSURL URLWithString:API_PREFIX relativeToURL:queryURL];
-    return queryURL;
++ (GADArticle *) articleFromDictionary: (NSDictionary*)dict {
```

Comment:
Can you make a ticket to create a `articleToDictionary` method that "undoes" this action? It will be useful in the future when we want to send updated articles to the server.